File: Backend_DotNet/WheelShare_dotNet/WheelShare_dotNet/Models/SavedPlace.cs
using System.ComponentModel.DataAnnotations;

namespace WheelShare_dotNet.Models
{
    public class SavedPlace
    {
        [Key]
        public int SavedPlaceId { get; set; }

        public int UserId { get; set; }

        public string RideName { get; set; } = string.Empty;

        // Pickup Location
        public string PickupAddress { get; set; } = string.Empty;
        public double PickupLatitude { get; set; }
        public double PickupLongitude { get; set; }

        // Drop Location
        public string DropAddress { get; set; } = string.Empty;
        public double DropLatitude { get; set; }
        public double DropLongitude { get; set; }

        public DateTime CreatedAt { get; set; }

        // Navigation property
        public User User { get; set; } = null!;
    }
}

File: Backend_Springboot/cab-booking/src/main/java/com/wheelshare/cabbooking/entity/SavedPlace.java
package com.wheelshare.cabbooking.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Entity
@Data
@Table(name = "saved_places")
public class SavedPlace {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer savedPlaceId;

    @Column(name = "user_id", nullable = false)
    private Integer userId;

    private String rideName;

    // Pickup Location
    private String pickupAddress;
    private Double pickupLatitude;
    private Double pickupLongitude;

    // Drop Location
    private String dropAddress;
    private Double dropLatitude;
    private Double dropLongitude;

    private LocalDateTime createdAt = LocalDateTime.now();

    // Navigation property
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", insertable = false, updatable = false)
    private User user;
}
